shell/src/main/java/alluxio/cli/fs/command/UpdateMountCommand.java

```diff
@@ -0,0 +1,120 @@
+import alluxio.exception.status.InvalidArgumentException;
+import alluxio.grpc.MountPOptions;
+
+import com.google.common.collect.Maps;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+
+import java.io.IOException;
+import java.util.Properties;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Update options for an Alluxio mount point.
+ */
+@ThreadSafe
+public final class UpdateMountCommand extends AbstractFileSystemCommand {
+
+  private static final Option READONLY_OPTION =
+      Option.builder()
+          .longOpt("readonly")
+          .required(false)
+          .hasArg(false)
+          .desc("mount point is readonly in Alluxio")
+          .build();
+  private static final Option SHARED_OPTION =
+      Option.builder()
+          .longOpt("shared")
+          .required(false)
+          .hasArg(false)
+          .desc("mount point is shared")
+          .build();
+  private static final Option OPTION_OPTION =
+      Option.builder()
+          .longOpt("option")
+          .required(false)
+          .hasArg(true)
+          .numberOfArgs(2)
+          .argName("key=value")
+          .valueSeparator('=')
+          .desc("options for this mount point. For security reasons, no options from existing mount"
+              + " point will be inherited.")
+          .build();
+
+  /**
+   * @param fsContext the filesystem of Alluxio
+   */
+  public UpdateMountCommand(FileSystemContext fsContext) {
+    super(fsContext);
+  }
+
+  @Override
+  public String getCommandName() {
+    return "updateMount";
+  }
+
+  @Override
+  public Options getOptions() {
+    return new Options().addOption(READONLY_OPTION).addOption(SHARED_OPTION)
+        .addOption(OPTION_OPTION);
+  }
+
+  @Override
+  public int run(CommandLine cl) throws AlluxioException, IOException {
+    String[] args = cl.getArgs();
+    AlluxioURI alluxioPath = new AlluxioURI(args[0]);
+    MountPOptions.Builder optionsBuilder = MountPOptions.newBuilder();
+
+    if (cl.hasOption(READONLY_OPTION.getLongOpt())) {
+      optionsBuilder.setReadOnly(true);
+    }
+    if (cl.hasOption(SHARED_OPTION.getLongOpt())) {
+      optionsBuilder.setShared(true);
+    }
+    if (cl.hasOption(OPTION_OPTION.getLongOpt())) {
+      Properties properties = cl.getOptionProperties(OPTION_OPTION.getLongOpt());
+      optionsBuilder.putAllProperties(Maps.fromProperties(properties));
+    }
+    mFileSystem.updateMount(alluxioPath, optionsBuilder.build());
+    System.out.println("Updated mount point options at " + alluxioPath);
+    return 0;
+  }
+
+  @Override
+  public String getUsage() {
+    return "updateMount [--readonly] [--shared] [--option <key=val>] <alluxioPath>";
+  }
+
+  @Override
+  public String getDescription() {
+    return "Updates options for a mount point.\n"
+        + "WARNING: updating mount point with incorrect options may cause mount point to"
+        + " become unusable or unmounted.";
```

Comment:
This seems like a very steep cost for using the command, given that there is no option to dry-run it

Reply:
I already added a dry run within this command (on the server side), so 99% of the time the old mount point will continue to work even if a user put something invalid in the command. But given the possibilities this command enables (we don't have control over what options they updates and that can be any valid HDFS configuration property), a user can still potentially make the UFS client malfunction in a way that even the dry run cannot detect. This is just to make sure they understand the potential impact of this command before running it.